src/commands/synthetics/run-tests-lib.ts

```diff
@@ -192,52 +168,117 @@ const getTestListBySearchQuery = async (
   }))
 }
 
-export const getTestsList = async (
+export const getFinalVersionOfTestsFromCi = async (
   api: APIHelper,
   config: RunTestsCommandConfig,
   reporter: MainReporter,
-  suites: Suite[] = []
+  suites?: Suite[]
 ) => {
-  // If "testSearchQuery" is provided, always default to running it.
-  if (config.testSearchQuery) {
-    const testsToTriggerBySearchQuery = await getTestListBySearchQuery(api, config.global, config.testSearchQuery)
+  let testConfig: TriggerConfig[]
+  // TODO: Clean up locations as part of SYNTH-12989
+  const configFromEnvironment = config.locations?.length ? {locations: config.locations} : {}
+
+  let testsToTrigger = config.publicIds.map((id) => ({
+    config: {
+      ...config.global,
+      ...configFromEnvironment,
+    },
+    id,
+  }))
 
-    if (testsToTriggerBySearchQuery.length > MAX_TESTS_TO_TRIGGER) {
-      reporter.error(
-        `More than ${MAX_TESTS_TO_TRIGGER} tests returned by search query, only the first ${MAX_TESTS_TO_TRIGGER} will be fetched.\n`
+  if (!testsToTrigger.length && config.testSearchQuery) {
+    try {
+      testsToTrigger = await getTestsFromSearchQuery(api, config, reporter)
+      testsToTrigger = testsToTrigger.map((test) => ({
+        ...test,
+        config: {
+          ...configFromEnvironment,
+          ...test.config,
+        },
+      }))
+    } catch (error) {
+      throw new CriticalError(
+        isForbiddenError(error) ? 'AUTHORIZATION_ERROR' : 'UNAVAILABLE_TEST_CONFIG',
+        error.message
       )
     }
+  }
 
-    return testsToTriggerBySearchQuery
+  try {
+    testConfig = await getTestConfig(config, reporter, suites)
+  } catch (error) {
+    throw new CriticalError(isForbiddenError(error) ? 'AUTHORIZATION_ERROR' : 'UNAVAILABLE_TEST_CONFIG', error.message)
+  }
+
+  if (!testsToTrigger.length) {
+    testsToTrigger = testConfig.map((test) => ({
+      ...test,
+      config: {
+        ...config.global,
+        ...configFromEnvironment,
+        ...test.config,
+      },
+    }))
+  } else {
+    testsToTrigger = overrideWithTestConfig(testsToTrigger, testConfig)
+  }
+
+  if (!testsToTrigger.length) {
+    throw new CiError('NO_TESTS_TO_RUN')
+  }
+
+  return testsToTrigger
+}
+
+const getTestsFromSearchQuery = async (api: APIHelper, config: RunTestsCommandConfig, reporter: MainReporter) => {
+  const testsToTriggerBySearchQuery = await getTestListBySearchQuery(api, config.global, config.testSearchQuery || '')
+
+  if (testsToTriggerBySearchQuery.length > MAX_TESTS_TO_TRIGGER) {
+    reporter.error(
+      `More than ${MAX_TESTS_TO_TRIGGER} tests returned by search query, only the first ${MAX_TESTS_TO_TRIGGER} will be fetched.\n`
+    )
   }
 
+  return testsToTriggerBySearchQuery
+}
+
+const getTestConfig = async (
```

Comment:
The notion of `TestConfig` is getting a bit confusing. What do you think of this naming?
```suggestion
const getTestConfigOverrideFromTestFiles = async (
```

Reply:
I like it, there's a card later on (https://datadoghq.atlassian.net/browse/SYNTH-12972) to rename `config` from the `TestConfig` to `testOverrides`, i think this renaming might be better as part of that work though 🤔 will add it as a DoD 👍 
For now i would prefer to keep the consistency 👍